src/include/sof/audio/audio_stream.h

```diff
@@ -525,6 +527,7 @@ static inline int
 audio_stream_bytes_without_wrap(const struct audio_stream *source,
 				const void *ptr)
 {
+	assert((intptr_t)source->end_addr >= (intptr_t)ptr);
```

Comment:
These should be separate commits too. Preferably errors

Reply:
Pretty sure these asserts are buggy, see demo in https://github.com/thesofproject/sof/pull/7929#discussion_r1264423943